pytorch_lightning/accelerators/gpu.py

```diff
@@ -45,12 +45,9 @@ def setup_environment(self) -> None:
 
     def setup(self, trainer: "pl.Trainer") -> None:
         self.set_nvidia_flags(trainer.local_rank)
-        return super().setup(trainer)
-
-    def on_train_start(self) -> None:
-        super().on_train_start()
         # clear cache before training
         torch.cuda.empty_cache()
+        return super().setup(trainer)
```

Comment:
the original order is call super first then empty_cache?
This will matter more soon as Adrian is working on moving accelerator into ttp, the setup function will merge then.

Reply:
good call, fixed. also removed the "return" as I don't think we need to return anything